rsconnect/bundle.py

```diff
@@ -378,56 +388,16 @@ def make_source_manifest(
     env_management_py: Optional[bool] = None,
     env_management_r: Optional[bool] = None,
 ) -> ManifestData:
-    manifest: ManifestData = cast(ManifestData, {"version": 1})
-
-    # When adding locale, add it early so it is ordered immediately after
-    # version.
-    if environment:
-        manifest["locale"] = environment.locale
-
-    manifest["metadata"] = {
-        "appmode": app_mode.name(),
-    }
-
-    if entrypoint:
-        manifest["metadata"]["entrypoint"] = entrypoint
-
-    if quarto_inspection:
-        manifest["quarto"] = {
-            "version": quarto_inspection.get("quarto", {}).get("version", "99.9.9"),
-            "engines": quarto_inspection.get("engines", []),
-        }
-
-        files_data = quarto_inspection.get("files", {})
-        files_input_data = files_data.get("input", [])
-        if len(files_input_data) > 1:
-            manifest["metadata"]["content_category"] = "site"
-
-    if environment:
-        package_manager = environment.package_manager
-        manifest["python"] = {
-            "version": environment.python,
-            "package_manager": {
-                "name": package_manager,
-                "version": getattr(environment, package_manager),
-                "package_file": environment.filename,
-            },
-        }
-
-    if image or env_management_py is not None or env_management_r is not None:
-        manifest["environment"] = {}
-        if image:
-            manifest["environment"]["image"] = image
-        if env_management_py is not None or env_management_r is not None:
-            manifest["environment"]["environment_management"] = {}
-            if env_management_py is not None:
-                manifest["environment"]["environment_management"]["python"] = env_management_py
-            if env_management_r is not None:
-                manifest["environment"]["environment_management"]["r"] = env_management_r
-
-    manifest["files"] = {}
-
-    return manifest
+    manifest: Manifest = Manifest(
```

Comment:
All places where we were manually creating `ManifestData` have been migrated to use the `Manifest` class. This ensures that the same behaviour is applied independently from the command used, and thus that the python version requirements are always applied to the manifest without having to set them in every single place.

This is also more robust to future changes to the manifest.